src/status_im/contexts/communities/actions/addresses_for_permissions/view.cljs

```diff
@@ -1,55 +1,76 @@
 (ns status-im.contexts.communities.actions.addresses-for-permissions.view
   (:require [quo.core :as quo]
+            [quo.foundations.colors :as colors]
             [react-native.core :as rn]
+            [reagent.core :as reagent]
             [status-im.common.not-implemented :as not-implemented]
             [status-im.contexts.communities.actions.addresses-for-permissions.style :as style]
             [utils.i18n :as i18n]
             [utils.re-frame :as rf]))
 
 (defn- account-item
-  [item]
-  [rn/view
-   {:style style/account-item-container}
-   [quo/account-avatar (assoc item :size 32)]
-   [rn/view
-    [quo/text
-     {:size   :paragraph-1
-      :weight :semi-bold}
-     (:name item)]
-    [quo/address-text
-     {:address (:address item)
-      :format  :short}]]])
+  [item _ _ selected-addresses]
+  [quo/account-permissions
+   {:account         {:name                (:name item)
+                      :address             (:address item)
+                      :emoji               (:emoji item)
+                      :customization-color (:customization-color item)}
+    :token-details   []
+    :checked?        (contains? @selected-addresses (:address item))
+    :on-change       (fn [checked?]
+                       (if checked?
+                         (swap! selected-addresses conj (:address item))
+                         (swap! selected-addresses disj (:address item))))
+    :container-style {:margin-bottom 8}}])
 
 (defn view
   []
   (let [{id :community-id}          (rf/sub [:get-screen-params])
         {:keys [name color images]} (rf/sub [:communities/community id])
-        accounts                    (rf/sub [:wallet/accounts-with-customization-color])]
-    [rn/safe-area-view {:style style/container}
+        accounts                    (rf/sub [:wallet/accounts-with-customization-color])
+        selected-addresses          (reagent/atom (rf/sub [:communities/addresses-for-permissions]))]
```

Comment:
Sure, I'll address this in a follow-up PR. I will also try to add unit tests along with it.

I couldn't find a similar implementation anywhere 😅. My rationale was that since the selected state is only used in this screen, it seemed more fitting to keep it local.

Reply:
No worries @ajayesivan I understand there's room for interpretation about when to use re-frame, and when not to use it. In my experience so far, it's easier to keep the UI simple & boring if we use re-frame as much as we can.

I think we can all agree that, in general (not always), it's a joy to work on UI code where there are no local state and only dispatches and subscriptions. Reagent atoms are definitely important, but should be used IMO only when strictly necessary, not the other way around. We lose so much when we move out of re-frame, e.g. harder to debug local state, less observability (no logs, no re-frisk), harder to unit test since the UI dictates more things, harder to write our integration tests because events require more arguments to be passed around, etc.

Reply:
Follow-up issue: https://github.com/status-im/status-mobile/issues/18405

Reply:
Hi, Thank you @Ajay for [linking](https://github.com/status-im/status-mobile/pull/18759#issuecomment-1935479896) to the discussion.

Wrapping a subscription inside a reagent atom seems wrong also to me. But making changes globally also doesn't look nice for the screen which has a confirm button. This causes weird scenarios/bugs like mentioned here: https://github.com/status-im/status-mobile/issues/18753#issuecomment-1933977943

I think the proper way would be to just pass the initial state as screen param while navigating to that screen and initialize the local state using that instead of subscription. @ilmotta @ajayesivan wdyt?

Reply:
Hi, Thank you @Ajay for [linking](https://github.com/status-im/status-mobile/pull/18759#issuecomment-1935479896) to the discussion.

Wrapping a subscription inside a reagent atom seems wrong also to me. But making changes globally also doesn't look nice for the screen which has a confirm button. This causes weird scenarios/bugs like mentioned here in PR description: https://github.com/status-im/status-mobile/pull/18759